pkg/mimirtool/client/backfill.go

```diff
@@ -0,0 +1,128 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package client
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/fs"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/pkg/errors"
+	"github.com/thanos-io/thanos/pkg/block/metadata"
+)
+
+func (c *MimirClient) Backfill(ctx context.Context, source string, logger log.Logger) error {
+	// Scan blocks in source directory
+	es, err := os.ReadDir(source)
+	if err != nil {
+		return errors.Wrapf(err, "failed to read directory %q", source)
+	}
+	for _, e := range es {
+		if err := c.backfillBlock(ctx, filepath.Join(source, e.Name()), logger); err != nil {
```

Comment:
Similar to server code, let's preconfigure logger with "user" field, so that we don't need to repeat it each time we use the logger.

Reply:
Done, PTAL.